exporter/internal/queue/queue.go

```diff
@@ -33,6 +33,10 @@ type Queue[T any] interface {
 	Size() int
 	// Capacity returns the capacity of the queue.
 	Capacity() int
+	// GetNextItem pulls the next available item from the queue along with its index. Once processing is
+	// finished, the index should be called with OnProcessingFinished to clean up the storage. If no new
+	// item is available, returns false.
+	GetNextItem() (uint64, T, bool)
```

Comment:
>If no new item is available, returns false.

Does it mean that it does return immediately? I think this is not the current implementation. See the other function comment "The function returns true when an item is consumed or false if the queue is stopped.". I think we need to make this blocking as well.

Reply:
Instead of exposing `getNextItem()` directly, I changed the PR to expose `Read()` instead. For persistent queue, `Read()` will be block until there is a new item in the queue or until the queue is stopped, while `getNextItem()` is only called when we are sure (from the channel) there is a new item in the queue, and thus is non-blocking.
Comment explaining `Read()` is in `queue.go`. What do you think?